assembly/src/assembler/instruction/procedures.rs

```diff
@@ -27,7 +27,12 @@ impl Assembler {
         context: &mut AssemblyContext,
     ) -> Result<Option<CodeBlock>, AssemblyError> {
         // make sure the procedure is in procedure cache
-        self.ensure_procedure_is_in_cache(proc_id, context)?;
+        let proc_name = context
+            .proc_map()
+            .get(proc_id)
+            .cloned()
+            .unwrap_or_else(|| panic!("procedure with ID {} is not found", proc_id));
+        self.ensure_procedure_is_in_cache(proc_id, &proc_name, context)?;
```

Comment:
Is there a reason to get `proc_name` here rather than inside `ensure_procedure_is_in_cache()` function? If we do it there, this will result in fewer code changes and also we won't need to get procedure name unless there is an actual error.